Armadillo/src/main/java/com/scribd/armadillo/playback/mediasource/HeadersMediaSourceHelperImpl.kt

```diff
@@ -0,0 +1,52 @@
+package com.scribd.armadillo.playback.mediasource
+
+import android.content.Context
+import com.google.android.exoplayer2.upstream.DataSource
+import com.google.android.exoplayer2.upstream.DefaultDataSource
+import com.google.android.exoplayer2.upstream.DefaultHttpDataSource
+import com.scribd.armadillo.Constants
+import com.scribd.armadillo.HeadersStore
+import com.scribd.armadillo.download.CacheManager
+import com.scribd.armadillo.models.AudioPlayable
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
```

Comment:
Just curious what `@Singleton` does here? I thought it only needs to be on the provides method.